### PR TITLE
Feature/hdf 68 alter safe halo transfer

### PR DESCRIPTION
**Pull request by ctverceles**
## Description of Changes

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-68)

Originally wrote the description of the JIRA task linked below to change the destination arg of the safeHaloTransfer function so that all unclaimed HALO rewards are sent to the HaloHalo (formerly HaloChest) contract instead of to the user -> but this is already [handled by other tasks](https://halodao.atlassian.net/browse/HDF-68?focusedCommentId=11475). I've just refractored some of the tests a bit more to try and make it more readable and at the same time get a sense of how the unit tests flow. 

Refractor changes
- attempted to "standardize" decimal precision, `parseFloat` and `formatEther` calls into reusable funcs to improve code readability
- applied the above to the unit tests to keep them passing
- moved `sleep` func to `./util/math.js`
- fixed typos in included PR template

## How To Test

1. run `npm i` at project root
2. then `npx hardhat test` (expect 42 tests to pass)

### Developer Checklist:

- [x] I have followed the guidelines in our Contributing document
- [x] This PR has a corresponding JIRA ticket
- [x] My branch conforms with our naming convention i.e. `feature/HDF-XXX-description`
- [x] I have written new tests for your core changes, as applicable
- [x] I have successfully ran tests locally

### Reviewers Checklist:

- [ ] Code is readable and understandable; any unclear parts have explanations
- [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
- [ ] I have successfully ran tests locally

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
-* [ ] This PR has a corresponding JIRA ticket
-* [ ] My branch conforms with our naming convention i.e. `feature/SOOW-XXX-description`
-* [ ] All files have appropriate file headers and documentation
-* [ ] I have written new tests for your core changes, as applicable
-* [ ] I have successfully ran tests locally
+- [x] I have followed the guidelines in our Contributing document
+- [x] This PR has a corresponding JIRA ticket
+- [x] My branch conforms with our naming convention i.e. `feature/HDF-XXX-description`
+- [x] I have written new tests for your core changes, as applicable
+- [x] I have successfully ran tests locally
```
Comment (tracyarciaga) [inline .github/pull_request_template.md:17]: Thanks for  correcting this.  If you don't mind, can you please add
-[] I have formatted my code using format document in VSCode.

**Review by tracyarciaga — APPROVED** [review]
Looks good, I just have  one  tiny request :)

**Review by tracyarciaga — CHANGES_REQUESTED** [review]
Please re-add .env.example

File: .github/pull_request_template.md (hunk 1 above)
Comment (ctverceles) [reply .github/pull_request_template.md:17]: Will do thanks @tracyarciaga !

**Review by tracyarciaga — APPROVED** [review]

**Pull request merged**